machine/v1beta1/types_azureprovider.go

```diff
@@ -183,6 +183,21 @@ type OSDisk struct {
 	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
 	// DiskSizeGB is the size in GB to assign to the data disk.
 	DiskSizeGB int32 `json:"diskSizeGB"`
+	// DiskSettings describe ephemeral disk settings for the os disk.
+	// +optional
+	DiskSettings DiskSettings `json:"diskSettings,omitempty"`
+	// CachingType specifies the caching requirements.
+	// Possible values include: 'None', 'ReadOnly', 'ReadWrite'. When empty defaults to 'None'.
+	// +optional
+	CachingType string `json:"cachingType,omitempty"`
+}
+
+// DiskSettings describe ephemeral disk settings for the os disk.
+type DiskSettings struct {
+	// EphemeralStorageLocation enables ephemeral OS when set to 'Local'.
+	// Possible values include: 'Local'.
```

Comment:
Indicate the meaning of empty is subject to change over releases

>  "" means no opinion and the platform chooses a default, which is subject to change over time, currently that default is blah

Also, add validation or build as enum string

Reply:
I'm not sure there is an option of this being empty right now, the field is not optional, if you set the DiskSettings struct in the upper struct, then you have to include `ephemeralStorageLocation: Local`, there is no other option.

The following isn't valid:
```
diskSettings:
  ephemeralStorageLocation: ""
```

How do we indicate this in terms of API validation? I think Enum with single value works, it's not marked as optional, is this correct?

I think we need to state here that this field is required when `DiskSettings` is present and the only valid value is `Local`?

Reply:
empty means no opinion, so you need to allow it so that "no opinion" works.